## **I. Problem Identification**

### **Background**

*Below is a fictious scenario aimed for data analysis practice*

Offbrand Inc. is a streetwear clothing brand based in the US that focuses on making clothes that are unique and authentic, aiming to redefine the standards of the fashion industry in the world. They have been selling their product through retail stores around the US for the last 5 years, and are looking to expand their operation through online e-commerce platform like Amazon. However to do this, they need to learn more about online shoppers purchasing behavior to come up with an effective online marketing and sales strategy.

### **SMART Goal**

- **Specific**: Identifying and analyzing trends for online shopper purchasing behavior, that includes factors influencing purchase decision, browsing habits, session length, and cart abandonment rates, to ultimately help with Offbrand's expansion to online stroes.
- **Measurable**: The online strategy informed by the dataset analysis will aim to generate 50,000 Amazon store page visits, 1,000 purchases, and achieve a conversion rate of 2%.
- **Achievable**: With the information available through the dataset analysis, Offbrand Inc. could smoothly transition from retail to online, as they already have prior experience in the industry.
- **Relevance**: To redefine the standards of the fashion industry in the world, the company needs to expand not only through physical store, but also in the digital space. So the goal is aligned with their vision.
- **Time-bound**: Finish the dataset analysis by the end of the week, followed by the launch of Offbrand Inc.'s online store by the end of the year.

### **User**

The Marketing and Sales Strategist of Offbranc Inc.

### **Problem Statement**

*Offbrand Inc. needs to analyze online shopper behavior to address gaps in purchasing trends, browsing habits, and cart abandonment, ensuring a data-driven strategy to achieve its online store goals by the end of the year.*

### **Breakdown of Problem**

**Problems for Data Visualization**:
- What is the monthly transaction trend of the online store?
    - Variable(s) used: Revenue, Month
- What is the correlation between bounce rates (percentage of people who leave after one page) and exit rates (percentage of people who leave after a series of pages)?
    - Variable(s) used: Bounce Rates, Exit Rates
- How frequent do users visit the website every month?
    - Variable(s) used: Month
- How much time does each user spend on respective page types? Which page type piques their interest the most?
    - Variable(s) used: Administrative Duration, Informational Duration, ProductRelated Duration

**Problems for Statistical Descriptive**:
- What is the central tendency of page values, and how do these values contribute to user transactions in average?
    - Variable(s) used: Page Value

**Problems for Statistical Inferential**:
- Do users spend more time on the online shop during special days compared to non-special days? 
    - Variable(s) used: Weekend, Revenue

## **II. Data Loading**

First, I will import the necessary module to load the data using Pandas.

In [2]:
# Import pandas module
import pandas as pd

# For data visualization
from scipy import stats
import plotly.express as px

Then, I will use pandas to read the csv file to put the data in a dataframe.

In [3]:
# Load csv into dataframe
df = pd.read_csv(r'C:\Users\fauza\Desktop\Data Software\master_hck021\p0-ftds021-hck-m1-fauazhima\online_shoppers_intention.csv')

# Checking whether the variable df has been loaded with data properly
df.head()

Administrative  Administrative_Duration  Informational  \
0               0                      0.0              0   
1               0                      0.0              0   
2               0                      0.0              0   
3               0                      0.0              0   
4               0                      0.0              0   

   Informational_Duration  ProductRelated  ProductRelated_Duration  \
0                     0.0               1                 0.000000   
1                     0.0               2                64.000000   
2                     0.0               1                 0.000000   
3                     0.0               2                 2.666667   
4                     0.0              10               627.500000   

   BounceRates  ExitRates  PageValues  SpecialDay Month  OperatingSystems  \
0         0.20       0.20         0.0         0.0   Feb                 1   
1         0.00       0.10         0.0         0.0   Feb                 2   
2         0.20       0.20         0.0         0.0   Feb                 4   
3         0.05       0.14         0.0         0.0   Feb                 3   
4         0.02       0.05         0.0         0.0   Feb                 3   

   Browser  Region  TrafficType        VisitorType  Weekend  Revenue  
0        1       1            1  Returning_Visitor    False    False  
1        2       1            2  Returning_Visitor    False    False  
2        1       9            3  Returning_Visitor    False    False  
3        2       2            4  Returning_Visitor    False    False  
4        3       1            4  Returning_Visitor     True    False

## **III. Data Cleaning**

### **Checking for Missing Values**

First, I need to check whether the data has any missing value.

In [4]:
# Checking for missing value
df.isnull().sum()

Administrative             0
Administrative_Duration    0
Informational              0
Informational_Duration     0
ProductRelated             0
ProductRelated_Duration    0
BounceRates                0
ExitRates                  0
PageValues                 0
SpecialDay                 0
Month                      0
OperatingSystems           0
Browser                    0
Region                     0
TrafficType                0
VisitorType                0
Weekend                    0
Revenue                    0
dtype: int64

### **Checking for Anomalies in Data Summary**

Then, I need to check the data summary to confirm whether there is an anomaly, such as incorrect data type and columns that aren't required to be investigated. 

In [5]:
# Displaying summary information.
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 12330 entries, 0 to 12329
Data columns (total 18 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   Administrative           12330 non-null  int64  
 1   Administrative_Duration  12330 non-null  float64
 2   Informational            12330 non-null  int64  
 3   Informational_Duration   12330 non-null  float64
 4   ProductRelated           12330 non-null  int64  
 5   ProductRelated_Duration  12330 non-null  float64
 6   BounceRates              12330 non-null  float64
 7   ExitRates                12330 non-null  float64
 8   PageValues               12330 non-null  float64
 9   SpecialDay               12330 non-null  float64
 10  Month                    12330 non-null  object 
 11  OperatingSystems         12330 non-null  int64  
 12  Browser                  12330 non-null  int64  
 13  Region                   12330 non-null  int64  
 14  TrafficType           

Here we can observe the following information:
- Regarding the dataset, it contains 12230 rows and 18 columns, which are broken down into 10 numeric columns and 8 descriptive columns.
- Regarding Data Type:
    - Columns that describe page type, such as Administrative, Informational, and ProductRelated, are of the correct data type integer, because it describes the amount of times the user has visited such data type in one session.
    - Columns that describe page type duration, such as Administrative_Duration, Informational_Duration, and ProductRelated_Duration, are of the correct data type float, because each session in each page is measured in seconds, and this dataset accounts for the miliseconds.
    - Columns that describe behavioral metrics of online shoppers, such as BounceRates, ExitRates, and PageValues, are of the correct data type float, because the rates are in percentage, and page values are in US Dollar.
    - Columns that describe the context of when the user accesses the site, such as SpecialDay, Month, dan Weekend, are of the correct respective data types. SpecialDay is float because the dataset describes it as this value takes a nonzero value between February 2 and February 12, zero before and after this date unless it is close to another special day. Month is integer because it describes **JUST** the month, not the specific datetime. Weekend is boolean because it describes whether the day the user accesses the site is in the weekends or not.
    - Columns that describe user information, such as OperatingSystems, Browser, Region, TrafficType, and VisitorType, are of the correct respective data types.
    - Revenue is of the correct data type, boolean because it detects whether the use has made a transaction or not.

Therefore, the data information shows no anomaly.

In [6]:
df['Month'].unique()

array(['Feb', 'Mar', 'May', 'Oct', 'June', 'Jul', 'Aug', 'Nov', 'Sep',
       'Dec'], dtype=object)

In [7]:
df['Month'] = df['Month'].replace({'June': 'Jun'})
df['Month'] = pd.to_datetime(df['Month'] + ' 2018', format='%b %Y')

### **Checking for Duplicates**

In [8]:
# Mencari baris duplikat
duplicates = df.duplicated()
print(f'Jumlah baris duplikat: {duplicates.sum()}')

Jumlah baris duplikat: 125


In [9]:
# Menghilangkan baris duplikat
df = df.drop_duplicates()

# Me-reset index agar index tetap konsisten dengan jumlah data
df = df.reset_index(drop=True)

# Mengecek kembali baris duplikat
duplicates = df.duplicated()
print(f'Jumlah baris duplikat: {duplicates.sum()}')

Jumlah baris duplikat: 0


In [10]:
# Membuat file csv dari dataframe yang kotor
df.to_csv('cleaned_df.csv', index=False)

**Conclusion: The data is almost CLEAN and requires only dropping duplicates.**

## **IV. Analysis and Calculation**

### **What is the monthly transaction trend of the online store?** (Data Visualisation)

This problem is aimed to analyze trend in transactions within the dataframe's period. 

First, we will create a new dataframe that displays the sum of revenue count each month.

In [12]:
# Creating a filtered dataframe to only consider value True from Revenue
transaction_count = df[df['Revenue'] == True]

# Creating another filtered dataframe that groups by month and counts the number of True in Revenue.
monthly_transaction_count = transaction_count.groupby('Month').size().reset_index(name='Transaction Count')

monthly_transaction_count

Month  Transaction Count
0 2018-02-01                  3
1 2018-03-01                192
2 2018-05-01                365
3 2018-06-01                 29
4 2018-07-01                 66
5 2018-08-01                 76
6 2018-09-01                 86
7 2018-10-01                115
8 2018-11-01                760
9 2018-12-01                216

Then, we will create a **line chart** from the dataframe that was created earlier. Here, we use line chart because we are trying to **identify trends in the number of transaction**, which is generally better represented through a line chart.

In [13]:
# Creating the bar chart with Month as the x-axis and Revenue Count as the y-axis
line_monthly_transaction_count = px.line(monthly_transaction_count, x='Month', y='Transaction Count', title='Monthly Transaction Count')

# Creating a filter on the datetime column to only display month
line_monthly_transaction_count.update_xaxes(tickformat='%b')

# Displaying the bar chart
line_monthly_transaction_count.show()

**INSIGHT**

The chart reveals a distinct seasonal trend, where there is a significant peak in November with a total transaction count of 760. This is likely due to an upcoming holiday sales event in the US like Black Friday, which is held by the end of November every year. In contrast, February and June exhibited the lowest points of transaction count in times where it really shouldn't, especially when June is holiday seasons. February's dip is likely because it is the time where the site had just launched, because the data did not start in January. While June might be due to an end to promotional campaigns in May. 

Aside from the peaks and the lows, we can see an increasing trend from February to May. The speculation behind this trend is after the end of February's hypothetical store opening, there are store opening promotional campaigns for online shoppers to incentivize using the online shop. Then hypothetically, the promotional campaign ended in June, which is probably associated with the end of the promotional campaign (usually it lasts for 3 months in general). Then, there is a consistent, steady rise in the data before and after the peak, which is synonymous to the organic growth of the online shop. 

These speculations, especially during the anomaly dips in February and June, would require further research for confirmation.

### **What is the correlation between bounce rates and exit rates?** (Data Visualisation)

Bounce rates are the rate of which a page causes a user to leave after viewing that singular page, and exit rates are the rate of which a page causes a user to leave after viewing multiple other pages before that. So the better performing page would be the one with least exit rate and bounce rate.

To explain the correlation between bounce rates and exit rates of each page in a user session, we need to use a **scatter plot**.

In [14]:
# Creating the scatter plot from dataframe, comparing Bounce Rates and Exit Rates
scatter_rates = px.scatter(df, x='BounceRates', y='ExitRates', title='Correlation Between Bounce Rates and Exit Rates')

# Displaying the scatter plot
scatter_rates.show()

**INSIGHTS**

The scatter plot shows a clear positive correlation, which means that pages with a higher tendency to lose users immediately also has a tendency to be the last page after the users have viewed other pages before. This implies the consistency at which each page is performing, where low-performing pages generally having both high bounce rates and exit rates, and high-performing pages generally having ow bounce rates and exit rates. The high density of data on the lower end (between 0 - 10%) suggests that these pages are engaging for users to not leave immediately.

There are, however, outliers in this data. There is a data on the uppermost left with 0% bounce rate and 20% exit rate, which indicates that no user ever stops on that page when opened directly, but only after visiting several other pages. This could mean that this page (as with any other page with 0% bounce rate and some amount of exit rate) serves as the ending of a transaction process, like confirmation page after purchase. There is also a data on the uppermost right with 20% bounce rate and 20% exit rate, which indicates an uninteresting website. These data points need to be investigated thoroughly.

### **How frequent do users visit the website every month?** (Data Visualisation)

Unlike monthly transaction frequency, this compares the amount of users that visit the website. This problem is aimed to analyze trend in users that enter the website within the dataframe's period. 

As with the first problem, trends are better displayed in a form of **line chart**. We will start off by creating a filtered data frame that only includes the amount of session, which is equal to the amount of rows.

In [16]:
# Creating another filtered dataframe that groups by month and counts the number of rows.
user_count = df.groupby('Month').size().reset_index(name='User Count')

user_count

Month  User Count
0 2018-02-01         181
1 2018-03-01        1860
2 2018-05-01        3329
3 2018-06-01         285
4 2018-07-01         432
5 2018-08-01         433
6 2018-09-01         448
7 2018-10-01         549
8 2018-11-01        2982
9 2018-12-01        1706

Then, we will create a **line chart** from the dataframe that was created earlier. Here, we use line chart because we are trying to **identify trends in the number of user count**, which is generally better represented through a line chart.

In [17]:
# Creating the bar chart with Month as the x-axis and Revenue Count as the y-axis
line_monthly_user_count = px.line(user_count, x='Month', y='User Count', title='Monthly User Session Count')

# Creating a filter on the datetime column to only display month
line_monthly_user_count.update_xaxes(tickformat='%b')

# Displaying the bar chart
line_monthly_user_count.show()

**INSIGHT**

The transaction count data and here is very similar in nature, but instead of peaking mostly in November, the highest peak can be seen in May with 3329 users checking out the site. This means that while transactions were mostly done in November, more users checked the website out during May. With this information, I could strengthen my hypothesis on the online store opening promo, where new users are more likely to check out the site than old users, which naturally means that there will be newer users closer to the opening than after almost a year. The second peak in November is also likely caused by the same reason: holiday seasons and Black Friday in the end of November.

Trends are also pretty much the same, with February until May rising very high due to my hypothetical situation, and a stable yet small rise from June due to the lack thereof. However, it can be noted that December has a slightly bigger proporton of users than the amount of transactions, likely through the remains of November holiday season promo, but also people not actually buying the products.

### **How much time does each user spend on respective page types? Which page type piques their interest the most?** (Data Visualisation).

This problem allows us to understand what each page type does and how do users react to each page type. The higher the duration of a page type, the more the user is interested on that specific page type. The three page types we are analyzing are:
- Administrative: Pages related to managing user accounts (like account registration, settings, etc.).
- Informational: Pages related to general informations related to the website (like about our company, contact, etc.).
- Product-Related: Pages related to information about products and product page themselves.

In [18]:
# Setting a new dataframe with the average of each page type duration
average_duration_per_page_type = df[['Administrative_Duration', 'Informational_Duration', 'ProductRelated_Duration']].mean().reset_index()

# Renaming the columns to it's respective purposes
average_duration_per_page_type.columns = ['Page Type', 'Average Duration in Seconds']

# Showing the dataframe to make sure it is correct
average_duration_per_page_type


Page Type  Average Duration in Seconds
0  Administrative_Duration                    81.646331
1   Informational_Duration                    34.825454
2  ProductRelated_Duration                  1206.982457

Here, we evaluate the difference using a **bar-chart**, which is perfect because we are just trying to compare numbers between all three categories.

In [19]:
# Creating a bar chart
fig = px.bar(average_duration_per_page_type, x='Page Type', y='Average Duration in Seconds',title='Average Time Spent on Different Page Types')

# Showing the bar chart
fig.show()

**INSIGHTS**

From the following chart, we can immediately see the elephant in the room: Product related page type with significantly higher time spent by user, which indicates that users are more engaged on pages related to products. This makes perfect sense since the whole point of an online store is to buy the products online. 

The other two page types are much less engaging. It makes sense for administrative page types to have less time spent, because these informations are only needed by users when they NEED to do it, and they will do it quickly. This implies while administrative page types are ESSENTIAL, but they do not caputre the users interest.

The page with the least amount of time spent by user (not by much) is informational page, where users do not feel like informations displayed in these pages are not compelling enough.

### **What is the average Page Values that lead to purchase? How does this compare to the one that does not lead to purchase?** (Descriptive Statistics)

Page value is simply a measure of how much a page is worth in terms of leading users up to transaction and eventually generating revenue. This question offers insights to the effectiveness different types of contents on the website in driving conversions. Here, we have to calculate the average page values that lead to transaction and that did not lead to transaction.

In [20]:
# Creating a filtered dataframe to only consider value True from Revenue
df_filter_true = df[df['Revenue'] == True]
df_filter_false = df[df['Revenue'] == False]

# Calculate the mean of each dataframe
print(f'Average page values that lead to transaction: ${df_filter_true['PageValues'].mean()}')
print(f'Average page values that do not lead to transaction: ${df_filter_false['PageValues'].mean()}')


Average page values that lead to transaction: $27.264518194696016
Average page values that do not lead to transaction: $1.9999853094621736


**INSIGHT**

Based on the average page values that was displayed above, we can conclude the following:
- For each user sessions that end up doing a transaction, they have clicked on **high-performing pages that lead to a generated revenue**. This suggests that these pages play an important role in guiding users to conversion. These pages could be the pages that are meant to finish in transaction, such as product information page, shopping cart, payment page, etc.
- In contrast, those who did not end up doing a transaction have mostly clicked on **low-performing pages that do not generate revenue**. This suggests that these pages lack the attractiveness for users to do a transaction. These pages could be the pages that are just meant to display information, such as home page, about me sections, etc.

### **Do users spend more time on the online shop during special days compared to non-special days?** (Inferential Statistics)

Because I am comparing between two samples (special day and non-special day), the most appropriate test to use is the **Two-Sample Independent Test**. 

In [21]:
# Creating a filtered dataframe to only consider value True from Revenue
df['TotalDuration'] = df['Administrative_Duration'] + df['Informational_Duration'] + df['ProductRelated_Duration']

In [22]:
# Assigning variables for special days and non-special days (in the data, value for special day is more than 0)
special_day_duration = df[df['SpecialDay'] > 0]['TotalDuration']
non_special_day_duration = df[df['SpecialDay'] == 0]['TotalDuration']

print(f'Average time spent on special day: {special_day_duration.mean()}')
print(f'Average time spent on non-special day: {non_special_day_duration.mean()}')

Average time spent on special day: 1018.9392068352826
Average time spent on non-special day: 1358.1694009928076


Based on the average time spent on special day and non-special days, the following hypothesis can be drawn.

> H0: Users spend less time during non-special days compared to special days.

> H1: Users spend more time during non-special days compared to special days.

To prove my hypothesis, I have to evaluate using t-staticstics and p-value.

In [23]:
# Testing t-statstics and p-value
t_stat, p_val = stats.ttest_ind(non_special_day_duration, special_day_duration)
print('T-Statistic:',t_stat)
print('P-value:',p_val)

T-Statistic: 5.564314776317639
P-value: 2.6873128996018925e-08


**INSIGHT**

Evaluating the results of t-stats (**5.56**) and p-value (**0.0000000268**), the following are the insights that could be concluded:
- Since the **t-Statistics** value is significantly larger than one, this indicates that the mean total time spent on online shop during special days is **significantly lower** than on non-special days.
- Since the **p-value** is very close to zero, and much less than the typical significance level of 0.05, this inidcates that the difference in time spent between the two variables are **statistically significant**.

With this evaluation, **H1 is proven**. This concludes that users do spend less time during special days than normal days. This could happen due to various reasons, such as during special days, people are more likely to spend less time on scrolling through online shops because they focus more on celebrating. While during non-special days, people are more likely to spend time browsing, because there is no occassion to celebrate.

### **Visualizations**

***[Click here!](https://public.tableau.com/views/OnlineShoppersPurchasingIntention_Dashboard/VisualizationDashboard?:language=en-US&publish=yes&:sid=&:redirect=auth&:display_count=n&:origin=viz_share_link)*** to view the organized visualizations with interactive features (powered by Tableau).

## **VI. Conclusion and Recommendation**

**CONCLUSION**

This analysis of online shopper purchasing intention dataset reveals patterns and trends that users tend to make in their journey of purchasing a product online. First, it highlights the significant peaks and dips in transaction counts and user sessions within this dataset's period, where we see a very similar but not the same trend. Then, we see a positive correlation between bounce rates and exit rates only between 0% to 20%, which means that all pages were consistent in their performance to keep users from exiting their page. Then, we learned that Product-Related pages are where users will end up on most of the time, which means that these pages are the KEY to conversions. Analysis also shows that non-special days are when people tend to check out online shops the most instead of special days. 

**RECOMMENDATIONS TO OFFBRAND INC.**
- **Enhancing Product-Related Content**: As product pages are where online users look the most for in an online shop, Offbrand Inc. should look to invest the most time optimizing these pages for their online store. 
- **In Promotions, Timing is Key**: Offbrand Inc. should always be on the look-out for the best time to schedule their promotions. Because as we can see, it does convert to user transaction and user sessions, especially in seasonal sales in the US like Black Friday.
- **Campaigns Should be Done in Low Points**: Why is this? Because dips are where people are least interested, Offbrand Inc. should learn from this dataset and create an effective campaign that sparks new interests when there isn't any, which in the long run will help stabilize revenue gains from even the lowest points of user transaction.
- **Exclusive Deals on Special Days**: People should have a reason to buy during Special Days, so that even during Special Days, users are still interested in visiting their website. This is best done by creating exclusive deals / products during Special Days.
- **Research on Outliers in Bounce Rates and Exit Rates**: By doing this, Offbrand Inc. will have a better understanding why some pages are clicked off more than others and improve based on that information.